src/components/Map/Map.jsx

```diff
@@ -1,44 +1,41 @@
 import React, { useEffect, useState } from 'react';
-import { MapContainer, TileLayer, Marker, Popup, useMap, ZoomControl } from 'react-leaflet';
+import { MapContainer, TileLayer, useMap, ZoomControl, Tooltip, Marker } from 'react-leaflet';
 import { GeoSearchControl, OpenStreetMapProvider } from 'leaflet-geosearch';
-import icons from 'leaflet-color-number-markers';
-import PopUpBox from '../PopUpBox/PopUpBox';
+import L from 'leaflet';
+import PropTypes from 'prop-types';
 
+import MarkerIcon from '../../assets/MarkerIcon.svg';
 import './Map.css';
+import { FYABackend } from '../../common/utils';
 
-const Map = () => {
-  // The activeMarker represents the current marker that is selected (i.e. which popup is shown)
-  const [activeMarker, setActiveMarker] = useState(null);
-  // This mapState variable stores the current instance of the map.
-  // This is used to fly to markers when they're clicked
+const Map = ({
+  setSelectedCountry,
+  setSelectedZipCode,
+  setSelectedBox,
+  setUpdateBoxListSwitch,
+  updateBoxListSwitch,
+}) => {
   const [mapState, setMapState] = useState(null);
+  // A list containing all unique zip codes stored in Anchor_Box
+  const [zipcodeData, setZipCodeData] = useState([]);
 
-  // This is sample data for Markers
-  // The only required pieces of data for Markers to show up are the long/lat coordinates
-  // Could use Nominatim to convert from city/state/etc name to coordinates (https://nominatim.org/release-docs/develop/api/Overview/)
-  const markerData = [
-    {
-      id: 'San Diego',
-      coordinates: [32.715736, -117.161087],
-      description: 'Sample marker at San Diego',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '1',
-    },
-    {
-      id: 'Irvine',
-      coordinates: [33.6846, -117.8265],
-      description: 'Sample marker at Irvine',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '2',
-    },
-    {
-      id: 'Los Angeles',
-      coordinates: [34.052235, -118.243683],
-      description: 'Sample marker at Los Angeles',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '3',
-    },
-  ];
+  // Handles when a marker is clicked
+  // 1. Updates the box list with the boxes located in the zip code (in PinInformation)
+  // 2. Switches PinInformation to box list view
+  const handleMarkerClicked = markerObject => {
+    // mapState.flyTo([markerObject.latitude, markerObject.longitude], 10);
+    setSelectedCountry(markerObject.country);
+    setSelectedZipCode(markerObject.zip_code);
+    // Toggle updateBoxListSwitch, which will update update the box list in the right side bar
+    setUpdateBoxListSwitch(!updateBoxListSwitch);
+    setSelectedBox(null);
+    mapState.flyTo([markerObject.latitude, markerObject.longitude], 10);
```

Comment:
Can we add a comment here saying that this call must be last in this function in order to avoid the pin moving bug.